src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -503,6 +515,180 @@ new qSelf qEnqueuePolicy qDequeuePolicy qFailurePolicy = liftIO $ do
     return OutQ{..}
 
 {-------------------------------------------------------------------------------
+  Monitoring the queue's health
+-------------------------------------------------------------------------------}
+
+-- | Queue health metrics
+data QHealth = QHealth {
```

Comment:
Why not EKG counters directly? We already depend on EKG directly in this module, and these are write-only counters right (unlike others that are stats used for enqueue/dequeue logic)?

Or if not EKG counters, why not IORefs?